Receiver_C/For_Testing/Test_case.cpp

```diff
@@ -0,0 +1,169 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+#include "catch.hpp"
+
+#include "Test_case.h"
+#include "../Common_Header.h"
+#include "../Data_Reader.h"
+#include "../Data_Analyzer.h"
+#include "../Output_Handler.h"
+#include "../Receiver_Main.h"
+
+TEST_CASE("Positive_case: Calculate SMA on given array") 
+{
+	double derived_SMA , data[5] = {1.0,3.0,5.0,7.0,9.0};	
+	double expected_SMA = 5.000000;
+    int retVal ; 	
+	retVal = Calculate_SMA( data , (sizeof(data)/(sizeof(data[0]))) , &derived_SMA );
+	REQUIRE( (retVal == OK) );
+	REQUIRE((derived_SMA == expected_SMA));		  
+}
+
+TEST_CASE("Negetive_case: Calculate SMA on given array") 
+{	
+    int retVal ; 	
+	retVal = Calculate_SMA( 0 , 0 , 0 );
+	REQUIRE(retVal == NOT_OK);		  
+}
+
+
+TEST_CASE("Positive_case: Find min and max on given array") 
+{
+	double min , max , data[5] = {1.0,3.0,5.0,7.0,9.0};	
+    int retVal ; 	
+	retVal = Min_Max_finder( data , (sizeof(data)/(sizeof(data[0]))) , &min , &max );
+	REQUIRE( (retVal == OK) );
+	REQUIRE((min == 1.000000));
+	REQUIRE((max == 9.000000));		  
+}
+
+TEST_CASE("Negetive_case: Find min and max on given array") 
+{	
+    int retVal ; 	
+	retVal = Min_Max_finder( 0 , 0 , 0 , 0);
+	REQUIRE(retVal == NOT_OK);		  
+}
+
+TEST_CASE("Test the Data analyzer function") 
+{	
+    int idx , Result = OK; 
+    char buffer[4][2] = { {'A' , TYPE_ALPHABET} , {'a' , TYPE_ALPHABET} , {'0' , TYPE_VALUE} , {'1' , TYPE_VALUE} };	
+	for(idx = 0; idx < 4 ; idx ++)
+	{
+		if( (Data_analyzer(buffer[idx][0])) != (buffer[idx][1]) )
+		{
+			Result = NOT_OK;
+		}	
+	}
+	REQUIRE(Result == OK);		  
+}
+
+TEST_CASE("Positive_case: Data storage validation") 

```

Comment:
Where is it storing the data? Maybe you mean something different with 'storage'. Do you mean 'streaming'?